packages/block/src/header.ts

```diff
@@ -354,8 +354,10 @@ export class BlockHeader {
     if (this._common.consensusAlgorithm() === ConsensusAlgorithm.Ethash) {
       // PoW/Ethash
       if (
+        number > BigInt(0) &&
```

Comment:
Ok, this disables genesis extradata check by default, I am not sure if we want this?

Reply:
This could be tricky since this check is in common and doesn't have access to the clients config flags that would indicate a test environment. We could have the hive simulator remove the extradata field if we're on PoW and that should circumvent the issue